File: Java/ejercicio3.java
import java.io.*;
import java.util.Scanner;

class Producto {
    private String codigo, nombre;
    private int cantidad;
    private double precio;

    public Producto(String codigo, String nombre, int cantidad, double precio) {
        this.codigo = codigo; this.nombre = nombre; this.cantidad = cantidad; this.precio = precio;
    }
    @Override
    public String toString() { return codigo + "," + nombre + "," + cantidad + "," + precio; }
}

public class ejercicio3 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int op;
        do {
            System.out.println("\n--- SISTEMA DE INVENTARIO ---");
            System.out.println("1. Registrar Producto\n2. Calcular Valor Total\n3. Salir");
            System.out.print("Seleccione: "); op = sc.nextInt(); sc.nextLine();

            if (op == 1) {
                System.out.print("Código: "); String cod = sc.next(); sc.nextLine();
                System.out.print("Nombre: "); String nom = sc.nextLine();
                System.out.print("Cantidad: "); int cant = sc.nextInt();
                System.out.print("Precio: "); double prec = sc.nextDouble();

                try (PrintWriter pw = new PrintWriter(new FileWriter("inventario.txt", true))) {
                    pw.println(new Producto(cod, nom, cant, prec).toString());
                    System.out.println("Producto guardado.");
                } catch (IOException e) { System.out.println("Error."); }
            } else if (op == 2) {
                File f = new File("inventario.txt");
                if (!f.exists()) { System.out.println("Sin registros."); continue; }
                double total = 0;
                try (BufferedReader br = new BufferedReader(new FileReader(f))) {
                    String l;
                    while ((l = br.readLine()) != null) {
                        String[] d = l.split(",");
                        total += (Integer.parseInt(d[2]) * Double.parseDouble(d[3]));
                    }
                    System.out.println("Valor total de activos en almacén: $" + total);
                } catch (Exception e) { System.out.println("Error de lectura."); }
            }
        } while (op != 3);
    }
}

File: C++/ejercicio3.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <sstream>

using namespace std;

class Producto {
private:
    string codigo, nombre;
    int cantidad;
    double precio;

public:
    Producto(string cod, string nom, int cant, double prec) {
        codigo = cod; nombre = nom; cantidad = cant; precio = prec;
    }
    string toString() {
        return codigo + "," + nombre + "," + to_string(cantidad) + "," + to_string(precio);
    }
};

void registrarProducto() {
    string cod, nom; int cant; double prec;
    cout << "Codigo del producto: "; cin >> cod;
    cin.ignore();
    cout << "Nombre del producto: "; getline(cin, nom);
    cout << "Cantidad: "; cin >> cant;
    cout << "Precio: "; cin >> prec;

    Producto prod(cod, nom, cant, prec);
    ofstream file("inventario.txt", ios::app);
    if (file.is_open()) {
        file << prod.toString() << endl;
        file.close();
        cout << "Producto añadido al inventario.\n";
    }
}

void calcularValorTotal() {
    ifstream file("inventario.txt");
    string linea;
    double granTotal = 0.0;

    if (file.is_open()) {
        while (getline(file, linea)) {
            if (linea.empty()) continue;
            stringstream ss(linea);
            string c, n, cantStr, precStr;
            getline(ss, c, ','); getline(ss, n, ',');
            getline(ss, cantStr, ','); getline(ss, precStr, ',');
            
            granTotal += (stoi(cantStr) * stod(precStr));
        }
        file.close();
        cout << "Valor total financiero del inventario: $" << granTotal << "\n";
    } else {
        cout << "No hay inventario registrado aún.\n";
    }
}

int main() {
    int op;
    do {
        cout << "\n--- INVENTARIO BÁSICO ---\n1. Registrar Producto\n2. Calcular Total Inventario\n3. Salir\nOpcion: ";
        cin >> op;
        if (op == 1) registrarProducto();
        if (op == 2) calcularValorTotal();
    } while (op != 3);
    return 0;
}
